pkg/rid/store/cockroach/store.go

```diff
@@ -48,33 +48,37 @@ type repo struct {
 // TODO: Add the SCD interfaces here, and collapse this store with the
 // outer pkg/cockroach
 type Store struct {
-	db     *cockroach.DB
-	logger *zap.Logger
-	clock  clockwork.Clock
+	db      *cockroach.DB
+	logger  *zap.Logger
+	clock   clockwork.Clock
+	version *semver.Version
 }
 
 // NewStore returns a Store instance connected to a cockroach instance via db.
 func NewStore(ctx context.Context, db *cockroach.DB, logger *zap.Logger) (*Store, error) {
+	vs, err := db.GetVersion(ctx, DatabaseName)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get database schema version for remote ID")
+	}
+
 	store := &Store{
-		db:     db,
-		logger: logger,
-		clock:  DefaultClock,
+		db:      db,
+		logger:  logger,
+		clock:   DefaultClock,
+		version: vs,
 	}
 
-	if err := store.CheckCurrentMajorSchemaVersion(ctx); err != nil {
+	if err := store.CheckCurrentMajorSchemaVersion(*vs); err != nil {
 		return nil, stacktrace.Propagate(err, "Remote ID schema version check failed")
 	}
 
 	return store, nil
 }
 
 // CheckCurrentMajorSchemaVersion checks that store supports the current major schema version.
-func (s *Store) CheckCurrentMajorSchemaVersion(ctx context.Context) error {
-	vs, err := s.GetVersion(ctx)
-	if err != nil {
-		return stacktrace.Propagate(err, "Failed to get database schema version for remote ID")
-	}
-	if vs == cockroach.UnknownVersion {
+func (s *Store) CheckCurrentMajorSchemaVersion(vs semver.Version) error {
```

Comment:
[Charlie's comment](https://github.com/interuss/dss/pull/401#discussion_r481489570) still applies here: there is no need to pass `vs` in because `version` is a pre-existing field in `Store`.  If we want to avoid dealing with nils, probably grabbing the version with `GetVersion` would be a good way to do things.  Or, we could make this a stand-alone function rather than a Store method.

Reply:
I changed the function back. The function now receives Context instead of version.